File: ios/GameFaceMatch/Core/GameAdapters/CollegeFootball27/CollegeFootball27Adapter.swift
import Foundation

struct CollegeFootball27Adapter: GameAppearanceAdapter {
    let catalogRepository: CatalogRepository
    let validator: CatalogValidator

    var gameID: String { "college-football-27" }
    var supportedVersions: [String] { [] }
    var supportedPlatforms: [String] { [] }

    init(catalogRepository: CatalogRepository, validator: CatalogValidator = CatalogValidator()) {
        self.catalogRepository = catalogRepository
        self.validator = validator
    }

    func validateCatalog(_ manifest: GameCatalogManifest) throws {
        try validator.validateProductionManifest(manifest, availableAssetIDs: [])
    }

    func match(profile: StandardFaceProfile) async throws -> [GameAppearanceMatch] {
        let manifest = try await catalogRepository.loadProductionManifest()
        try validateCatalog(manifest)
        guard manifest.items.isEmpty == false else {
            throw GameAdapterError.catalogUnavailable("Verified College Football 27 catalog not loaded.")
        }
        throw GameAdapterError.catalogUnavailable("College Football 27 matching is unavailable until production catalog matching is implemented.")
    }

    func buildInstructions(for match: GameAppearanceMatch) throws -> [BuildInstruction] {
        throw GameAdapterError.catalogUnavailable("Verified College Football 27 catalog not loaded.")
    }

    func refine(originalProfile: StandardFaceProfile, createdPlayerImages: [CapturedImage]) async throws -> RefinementResult {
        throw GameAdapterError.refinementUnavailable("Screenshot refinement is unavailable until verified catalog data exists.")
    }
}

File: web/lib/adapters/college-football-27-adapter.ts
import type { CatalogRepository } from "@/lib/catalog/catalog-repository";
import { validateProductionCatalog } from "@/lib/catalog/catalog-validator";
import { createRuleBasedMatchingEngine, type MatchingEngine, type MatchingPreferences } from "@/lib/matching/matching-engine";
import { CATALOG_UNAVAILABLE_MESSAGE } from "@/lib/product-copy";
import type { BuildInstruction, GameAppearanceMatch, GameCatalogManifest, RefinementResult, StandardFaceProfile } from "@/types/domain";
import { GameAdapterError, type GameAppearanceAdapter } from "./game-appearance-adapter";

export class CollegeFootball27Adapter implements GameAppearanceAdapter {
  readonly gameID = "college-football-27";
  readonly supportedVersions: string[] = [];
  readonly supportedPlatforms: string[] = [];

  constructor(
    private readonly catalogRepository: CatalogRepository,
    private readonly matchingEngine: MatchingEngine = createRuleBasedMatchingEngine()
  ) {}

  validateCatalog(manifest: GameCatalogManifest) {
    validateProductionCatalog(manifest);
  }

  async match(profile: StandardFaceProfile, preferences?: MatchingPreferences): Promise<GameAppearanceMatch[]> {
    const manifest = await this.catalogRepository.loadProductionManifest();
    this.validateCatalog(manifest);
    if (manifest.items.length === 0) {
      throw new GameAdapterError("catalogUnavailable", CATALOG_UNAVAILABLE_MESSAGE);
    }
    const matches = this.matchingEngine.matchTopThree({ profile, catalog: manifest, preferences, limit: 3 });
    if (matches.length === 0) {
      throw new GameAdapterError("catalogUnavailable", CATALOG_UNAVAILABLE_MESSAGE);
    }
    return matches;
  }

  buildInstructions(_match: GameAppearanceMatch): BuildInstruction[] {
    throw new GameAdapterError("catalogUnavailable", CATALOG_UNAVAILABLE_MESSAGE);
  }

  async refine(_originalProfile: StandardFaceProfile, _createdPlayerImages: File[]): Promise<RefinementResult> {
    throw new GameAdapterError("refinementUnavailable", "Screenshot refinement is unavailable until verified catalog data exists.");
  }
}
